ui/src/dashboards/apis/index.ts

```diff
@@ -1,102 +1,161 @@
-import AJAX from 'src/utils/ajax'
+// Libraries
+import _ from 'lodash'
 
-import {AxiosResponse} from 'axios'
-import {DashboardsResponse, GetDashboards} from 'src/types/apis/dashboards'
+// Utils
+import {addLabelDefaults} from 'src/shared/utils/labels'
+import {incrementCloneName} from 'src/utils/naming'
 
-export const getDashboards: GetDashboards = () => {
-  return AJAX<DashboardsResponse>({
-    method: 'GET',
-    resource: 'dashboards',
-  }) as Promise<AxiosResponse<DashboardsResponse>>
-}
+// Types
+import {Cell, NewCell, Dashboard, View} from 'src/types/v2'
+import {ILabel} from '@influxdata/influx'
 
-export const getDashboard = async dashboardID => {
-  try {
-    return await AJAX({
-      method: 'GET',
-      url: `/chronograf/v1/dashboards/${dashboardID}`,
-    })
-  } catch (error) {
-    console.error(error)
-    throw error
-  }
-}
+import {Cell as CellAPI, CreateDashboardRequest} from '@influxdata/influx'
+import {client} from 'src/utils/api'
 
-export const updateDashboard = dashboard => {
-  return AJAX({
-    method: 'PUT',
-    url: dashboard.links.self,
-    data: dashboard,
+export const addDashboardIDToCells = (
+  cells: CellAPI[],
+  dashboardID: string
+): Cell[] => {
+  return cells.map(c => {
+    return {...c, dashboardID}
   })
 }
 
-export const updateDashboardCell = cell => {
-  return AJAX({
-    method: 'PUT',
-    url: cell.links.self,
-    data: cell,
-  })
+// TODO(desa): what to do about getting dashboards from another v2 source
+export const getDashboards = async (): Promise<Dashboard[]> => {
+  const dashboards = await client.dashboards.getAll()
+
+  return dashboards.map(d => ({
+    ...d,
+    labels: d.labels.map(addLabelDefaults),
+    cells: addDashboardIDToCells(d.cells, d.id),
+  }))
 }
 
-export const createDashboard = async dashboard => {
-  try {
-    return await AJAX({
-      method: 'POST',
-      resource: 'dashboards',
-      data: dashboard,
-    })
-  } catch (error) {
-    console.error(error)
-    throw error
+export const getDashboard = async (id: string): Promise<Dashboard> => {
+  const dashboard = await client.dashboards.get(id)
+
+  return {
+    ...dashboard,
+    labels: dashboard.labels.map(addLabelDefaults),
+    cells: addDashboardIDToCells(dashboard.cells, dashboard.id),
   }
 }
 
-export const deleteDashboard = async dashboard => {
-  try {
-    return await AJAX({
-      method: 'DELETE',
-      url: dashboard.links.self,
-    })
-  } catch (error) {
-    console.error(error)
-    throw error
+export const createDashboard = async (
+  props: CreateDashboardRequest
+): Promise<Dashboard> => {
+  const dashboard = await client.dashboards.create(props)
+
+  return {
+    ...dashboard,
+    labels: dashboard.labels.map(addLabelDefaults),
```

Comment:
not for this pr, but perhaps we should consider doing this or having a function for this in the client library so we dont have to implement it everywhere.

Reply:
will do in next PR